### PR TITLE
Remove FactCheck, use Base.test and simplify tests

### PR DESCRIPTION
**Pull request by haampie**
This touches all tests, so a bit of a scary PR.

It removes `FactCheck` and moves to Julia's native `Base.Test`

It also simplifies a lot of tests and removes some warnings in 0.6 about broadcasting.

With regard to the `lsmr.jl` test, it's quite elaborate (the types & methods in the test itself almost need tests), but the actual thing that is being tested is whether or not lsmr support custom matrix and vector types. I think checking for custom matrix types can just as well be done by wrapping a matrix in a `LinearMap`, rather than writing hard to follow implicit matrix-vector product routines.

Concerning the custom vector types: that's something we need to test not just for `lsmr`, but for all methods. For instance using `DistributedVector`.

This PR also removes the test with the `test/IterativeSolvers.jld` which is basically a huge matrix (see #117).

Lastly, the tests in `history.jl` should probably not be smoke tests like they are now. I think it makes more sense to just unit test a `ConvergenceHistory` object.

### CODE REVIEW

**Comment by haampie** [comment]
It turns out I accidentally broke support for Julia 0.5 in multiple ways:

- The new version of LinearMaps.jl drops 0.6 support
- `@test [expr] [kwargs]` syntax is not backwards compatible
- The GMRES test with preconditioners and sparse matrices was broken, and after a fix does not pass on Julia 0.5. It accidentally converted sparse to dense via `sprand(...) + eye(...)`, which I have fixed to `sprand(...) + speye(...)`. And only then the tests failed on Julia 0.5 because `A_ldiv_B!(::UmfpackLU, ::SubArray)` was not implemented. Current solution is to do the test conditionally on the version number.

File: test/gmres.jl [hunk 1]
```diff
@@ -1,91 +1,65 @@
...
-    R = sprandn(n,n,0.5)
-    b = convert(Vector{T}, randn(n))
-    if T <: Complex
-        A += im*sprandn(n,n,0.5)
-        L += im*sprandn(n,n,0.5)
-        R += im*sprandn(n,n,0.5)
-        b += im*randn(n)
-    end
+@testset "SparseMatrixCSC{$T}" for T in (Float64, Complex128)
+    A = sprand(T, n, n, 0.5) + speye(T, n, n)
```
Comment (andreasnoack) [inline test/gmres.jl]: Use `I` here instead of `speye`

Comment (haampie) [inline test/gmres.jl]: Nice, I'll use that throughout the code then

File: test/cg.jl [hunk 2]
```diff
@@ -1,45 +1,44 @@
...
+        # All-zeros rhs should give all-zeros lhs
+        x0 = cg(A, zeros(T, n))
+        @test x0 == zeros(T, n)
+    end
 end
 
-context("Sparse Laplacian") do
-    A = poisson_matrix(Float64, 32, 3)
+@testset "Sparse Laplacian" begin
+    A = poisson_matrix(Float64, 10, 3)
```
Comment (andreasnoack) [inline test/cg.jl]: Why change the size?

Comment (haampie) [inline test/cg.jl]: At the moment is generates a matrix of size 32^3 = 32.768, which is rather large for a test.

File: src/cg.jl [hunk 3]
```diff
@@ -118,17 +118,15 @@ function cg_iterator!(x, A, b, Pl = Identity();
...
 
-    # Stopping criterion
-    ρ = one(residual)
-
     # Return the iterable
     if isa(Pl, Identity)
         return CGIterable(A, x, b,
             r, c, u,
-            reltol, residual, ρ,
+            reltol, residual, residual,
```
Comment (haampie) [inline src/cg.jl]: Whoops, yeah, that seems to be an error. I'll fix it
